### PR TITLE
Refactoring gyarados

### PR DESCRIPTION
**Pull request by CosmeValera**

### CODE REVIEW

File: src/modelo/Ficheros.java [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+    public static GUI GUICallBack;
+
+    public static void listadoPartidas(List<String> listadoNombreFicheros) throws Exception {
+
+        try ( FileInputStream fis = new FileInputStream("todosLosFicheros.txt");  DataInputStream ois = new DataInputStream(fis)) {
+            while (true) {
+                listadoNombreFicheros.add(ois.readUTF());
+            }
+        } catch (IOException e) {
+            //:D
```
Comment (Chemaclass) [inline src/modelo/Ficheros.java:27]: :D

File: src/modelo/Ficheros.java [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+    public static void guardarDataDatos(String nombreFichero) throws Exception {
+        File fichero = new File("partida_" + nombreFichero + ".txt");
+
+        try ( FileOutputStream fos = new FileOutputStream(fichero);  DataOutputStream oos = new DataOutputStream(fos)) {
+            oos.writeInt(Global.starter.getCantidadOro());
+            oos.writeInt(Global.starter.getEnemigosVencidos());
+            oos.writeBoolean(Starter.isTeemoDisponible());
+            oos.writeBoolean(Starter.isPoppyDisponible());
+            oos.writeBoolean(Starter.isGnarDisponible());
+            oos.writeBoolean(Starter.isYuumiDisponible());
```
Comment (Chemaclass) [inline src/modelo/Ficheros.java:97]: OSS? Open-Source Software? ;)

File: src/modelo/Global.java [hunk 3]
```diff
@@ -2,12 +2,13 @@
...
 import javax.swing.JPanel;
 
-public class Global {
+public class Global{
 
     public static Starter starter;
     public static Teemo teemo;
     public static Gnar gnar;
     public static Poppy poppy;
+    public static Yuumi yuumi;
```
Comment (Chemaclass) [inline src/modelo/Global.java:11]: 😋

File: src/modelo/Gnar.java [hunk 4]
```diff
@@ -2,7 +2,7 @@
 
 public class Gnar extends Starter {
 
-    private int resistencia;
+    private int resistencia; //PARA FICHEROS
```
Comment (Chemaclass) [inline src/modelo/Gnar.java]: Is this comment relevant? What about expressing this intention in the variable name instead?

File: src/modelo/Gyarados.java [hunk 5]
```diff
@@ -2,6 +2,8 @@
 
 public class Gyarados extends Enemigo {
 
+    private double resistenciaMagica = 55; //Ligeramente fuerte contra magia
```
Comment (Chemaclass) [inline src/modelo/Gyarados.java]: Commentarios... 🥉

File: src/modelo/Pikachu.java [hunk 6]
```diff
@@ -2,10 +2,12 @@
 
 public class Pikachu extends Enemigo {
 
+    private double resistenciaMagica = 35; //Debil contra magia
```
Comment (Chemaclass) [inline src/modelo/Pikachu.java]: Again "...contra magia". What about creating some constants which express this intention?

**Review by Chemaclass — APPROVED** [review]
Overall it's great 👍

File: src/modelo/Gnar.java (hunk 4 above)
Comment (CosmeValera) [reply src/modelo/Gnar.java]: It isn't. I'll change it

**Pull request merged**